Checks ssl

dustinleblanc opened the pull request:
@TeslaDethray I thought it might be nice to let Terminus directly tell folks about the SSL version issue rather than have them have to come to support for this. This of course only applies when TLS 1.0 is removed from Terminus. Thoughts on this? I'm not wild by the use of statics, but it seemed appropriate given the context.

Thanks!

dustinleblanc committed:
Remove commented code

dustinleblanc committed:
Add Requirements check helper.

dustinleblanc committed:
Validate the Terminus runtime environment meets requirements.

pantheon-circleci commented:
@dustinleblanc, thanks for your PR! By analyzing the blame information on this pull request, we identified @TeslaDethray to be a potential reviewer

dustinleblanc committed:
Allow auto-fix from linter

SaraMcCutcheon commented:
This is great, @dustinleblanc! I was in a meeting yesterday when where were were discussing some very similar work to this and related topics. (https://getpantheon.atlassian.net/browse/AL-392) You've just completed a large portion of the Terminus work to be done on this matter! <3

I have a few concerns about the PR as it is right now:
- The linter errors
- The PHP version is already enforced by Composer, yes?
- We already have an "UpdateHelper" which is very similar to this. Perhaps they both should be renamed or joined?
- Because Helpers are accessed by the commands, we can't enforce its use for users who use Terminus as a library (i.e. they use the models and collections and skip the CLI entirely or they've made a plugin).

dustinleblanc commented:
Hey @TeslaDethray so I fixed the linter issue, so that's no longer a problem. I also added a `-f` option flag to our linter script so it can just do it for you if you'd like.

The PHP version checker is a dupe yeah, I'd be okay removing it. we have another check int he boot file too that we might be able to remove also?

I'll take a look at updatehelper, the name led me to believe it did different tasks. Thanks for the feedback!

dustinleblanc committed:
Combine RequirementsHelper and UpdateHelper

dustinleblanc commented:
Closing as Master as moved on far, far away from this :)
